### DISCUSSION TITLE
Show CBF Version

### DISCUSSION THREAD

**Discussion by N0cynym** [jglim/CaesarSuite#9] [Ideas]
### The Idea
Showing the CBF_Version in Diogenes for better bug fixing

#### Example

EZS | Version | Date | Size
-- | -- | -- | --
EIS_164.cbf | 02.00.01 | 10.05.2012 | 396kB
EIS_204.cbf | 04.00.11 | 30.10.2012 | 350kB
EIS_212.cbf | 03.01.19 | 26.03.2014 | 258kB
EIS166.cbf | 02.03.22 | 18.05.2015 | 636kB
EIS221.cbf | 03.00.01 | 10.05.2012 | 516kB
EIS222.cbf | 02.11.09 | 05.10.2015 | 2379kB
EZS1_H.cbf | 00.00.00 | 16.02.2012 | 12kB
EZS164.cbf | 01.07.22 | 16.02.2012 | 261kB
EZS166.cbf | 04.04.02 | 15.06.2017 | 757kB
EZS169.cbf | 01.03.37 | 10.05.2012 | 414kB
EZS203.cbf | 04.11.10 | 16.02.2012 | 236kB
EZS209.cbf | 02.10.05 | 11.05.2012 | 1485kB
EZS211.cbf | 04.04.03 | 10.05.2012 | 760kB
EZS218.cbf | 04.03.15 | 22.09.2015 | 368kB
EZS220.cbf | 06.00.13 | 16.02.2012 | 139kB
EZS221.cbf | 01.07.05 | 16.02.2012 | 380kB
EZS230.cbf | 03.16.03 | 16.02.2012 | 563kB
EZS231.cbf | 02.01.07 | 05.04.2014 | 552kB

**Comment by jglim** [comment]
The only version and date I can find is in the container's CFF header, which looks like this:

```
CRD3:
CbfVersionString : ((VERSION 04.00.01, CBF -- generated by SP based CTRAFO 03.00.14))
GpdVersionString : ((VERSION 03.77.02, 18.01.06, GPD-Translator))

MED40:
CbfVersionString : ((VERSION 04.00.01, CBF -- generated by SP based CTRAFO 03.00.15))
GpdVersionString : ((VERSION 03.77.02, 18.01.06, GPD-Translator))

VGSNAG2:
CbfVersionString : ((VERSION 04.00.01, CBF -- generated by SP based CTRAFO 03.00.14))
GpdVersionString : ((VERSION 03.77.02, 18.01.06, GPD-Translator))
```

The suggestion is valid, but I can't seem to find the data that is referenced in the above table, so I might not be capable enough to make it work. If the file has to be uniquely identified, maybe the checksum would be enough.

**Reply by N0cynym** [reply]
I don't know which version your's `vgsnag.cbf` but in my case the cbf-version is written at adress `0x000004FC` 8 bytes = `04.00.95`.

**Reply by N0cynym** [reply]
If you're using [npp](https://notepad-plus-plus.org/) go to line 11.

**Reply by jglim** [reply]
Got it, the actual value is under `CaesarContainer->ECU->EcuXmlVersion`. I'll think of a place to show the version without adding too much clutter.

**Reply by jglim** [reply]
Added in https://github.com/jglim/CaesarSuite/commit/4f3488226c3dc93eee4e2c19206867f4e693b21d
